eng/tools/mcp/azure-sdk-java-mcp/src/java-sdk-changelog.ts

```diff
@@ -0,0 +1,133 @@
+import { CallToolResult } from "@modelcontextprotocol/sdk/types";
+import { findAzureSdkRoot, spawnAsync } from "./utils/index.js";
+import { mkdtemp, rm } from "fs/promises";
+import { join } from "path";
+import { tmpdir } from "os";
+import axios from "axios";
+import { XMLParser } from "fast-xml-parser";
+import fs from "fs";
+
+const MAVEN_HOST = "https://repo1.maven.org/maven2/";
+
+export async function getJavaSdkChangelog(
+    jarPath: string,
+    groupId: string,
+    artifactId: string,
+): Promise<CallToolResult> {
+    let tempDir: string | null = null;
+
+    try {
+        // make sure cwd is the root directory of the Azure SDK for Java
+        let rootDirectory = await findAzureSdkRoot(process.cwd());
+        const changelogJson = await getJavaSdkChangelogJson(rootDirectory, jarPath, groupId, artifactId);
+        return {
+            content: [
+                {
+                    type: "text",
+                    text: changelogJson ? changelogJson.changelog : "No changelog",
+                },
+            ],
+        };
+    } catch (error) {
+        return {
+            content: [
+                {
+                    type: "text",
+                    text: `Unexpected error during SDK changelog: ${error instanceof Error ? error.message : String(error)}`,
+                },
+            ],
+        };
+    } finally {
+        // Clean up temporary directory
+        if (tempDir) {
+            try {
+                await rm(tempDir, { recursive: true, force: true });
+            } catch (cleanupError) {
+                console.error(`Failed to clean up temporary directory: ${tempDir}`, cleanupError);
+            }
+        }
+    }
+}
+
+export interface Changelog {
+    changelog: string;
+    breakingChanges: string[];
+}
+
+export async function getJavaSdkChangelogJson(
+    repoRoot: string,
+    jarPath: string,
+    groupId: string,
+    artifactId: string,
+): Promise<Changelog | undefined> {
```

Comment:
Can we simplify this? It's hard for users to specify all of these to just get the changelog. Can we get to the changelog given just the artifact id? Why is jarPath needed? This would be then make building the library a prerequisite to have the jar path.

Reply:
This function is not a tool. It is just shared code.

But yes, currently this tool requires build package first. My justification is that it makes tool modular (agent can figure out the JAR path, if it is built). Though we can just change it to have it call "build" first (again, some shared code), then calculate.